tari-project/universe
# fix: handle missing microsoft visual c++ redistributable ( fix part 1/2 )

### [ Features WINDOWS ONLY ]
- Implemented reading of installed applications on system
- Added list of required external dependencies on system
- Added detection if user have all required dependecies
- Improved error handling to show what dependency is missing with link to download page
------------
### [ Notes ]
From what I've been testing gpu miner:
- Will work fine with `2022` and `2019` packages
- Throw `non-zero exit code Some(-1073741515` for versions `2017` and `2015`

Going with lower versions like `2012` will cause to app not starting at all and throwing error about missing msvcp140.dll which comes with versions `2015` and higher

------------
### [ Testing ]

- Remove all Microsoft Visual C++ Redistributable packages from system
- Install for example Microsoft Visual C++ Redistributable 2015 https://www.microsoft.com/pl-pl/download/details.aspx?id=48145
- After opening application you should see information about missing packages
![image](https://github.com/user-attachments/assets/4b9670e0-057b-4c75-a938-f8459c320a53)
- After installing required versions application should continiue normally
------------
### [ Issues ]
- On my machine I found that without any Microsoft Visual C++ Redistributable package I cannot even start application, getting error of missing msvcp140.dll
